src/quo2/components/settings/data_item/view.cljs

```diff
@@ -0,0 +1,141 @@
+                     :source              (resources/tokens :eth)
+                     :style               style/image}]
+          :default nil
+          nil)])
+     [text/text
+      {:weight :medium
+       :size   :paragraph-2
+       :style  (style/description blur? theme)}
+      subtitle]]))
+
+(defn- left-title
+  [{:keys [title label size theme]}]
+  [rn/view {:style style/title-container}
+   [text/text
+    {:weight :regular
+     :size   :paragraph-2
+     :style  (style/title theme)}
+    title]
+   (when (and (= :graph label) (not= :small size))
+     [text/text
+      {:weight :regular
+       :size   :label
+       :style  (style/title theme)}
+      (i18n/label :t/days)])])
+
+(defn- left-side
+  [{:keys [theme title status size blur? description icon subtitle label icon-color emoji-color emoji]}]
+  [rn/view {:style style/left-side}
+   [left-title
+    {:title title
+     :label label
+     :size  size
+     :theme theme}]
+   (if (= status :loading)
+     [left-loading
+      {:size  size
+       :blur? blur?
+       :theme theme}]
+     [left-subtitle
+      {:theme       theme
+       :size        size
+       :description description
+       :icon        icon
+       :icon-color  icon-color
+       :blur?       blur?
+       :subtitle    subtitle
+       :emoji-color emoji-color
+       :emoji       emoji}])])
+
+(defn- right-side
+  [{:keys [label icon-right? icon-color]}]
+  [rn/view {:style style/right-container}
+   (case label
+     :preview [preview-list/preview-list
+               {:type      :user
+                :size      24
+                :list-size 3}]
+     :graph   [text/text "graph"]
+     :none    nil
+     nil)
+   (when icon-right?
+     [rn/view {:style (style/right-icon label)}
+      [icons/icon
+       (if (= :none label)
+         :i/copy
+         :i/chevron-right)
+       {:accessibility-label :icon-right
+        :color               icon-color
+        :size                20}]])])
+
+(def view-internal
+  (fn [{:keys [blur? card? icon-right? label description status size theme on-press title subtitle
+               icon emoji-color emoji]}]
+    (let [icon-color (if (or blur? (= :dark theme))
+                       colors/white
+                       colors/neutral-100)]
+      (if (= :graph label)
+        [not-implemented/view {:blur? blur?}]
+        [rn/pressable
+         {:accessibility-label :data-item
+          :disabled            (not icon-right?)
+          :on-press            on-press
+          :style               (style/container size card? blur? theme)}
+         [left-side
+          {:theme       theme
+           :title       title
+           :status      status
+           :size        size
+           :blur?       blur?
+           :description description
+           :icon        icon
+           :subtitle    subtitle
+           :label       label
+           :icon-color  icon-color
+           :emoji-color emoji-color
+           :emoji       emoji}]
```

Comment:
You can alias this map and pass it to avoid passing every single prop independently:

```clojure
(defn my-component 
  [{:keys [a b c d] :as my-alias}] ;; give a name to the whole map
  [rn/view
   ,,,
   [my-component-2 my-alias]]) ;; pass the map
```

Or if you need to assoc new keys:

```clojure
(defn my-component
  [{:keys [a b c d] :as my-alias}] ;; give a name to the whole map
  [rn/view
  ,,,
   [my-component-2 (assoc my-alias :x "X")]]) ;; pass it adding an extra key
```

Reply:
Nice 👍

Reply:
Done